src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs

```diff
@@ -434,5 +434,36 @@ class App
             Assert.Null(comp.Assembly.GetTypeByMetadataName("System.Threading.Tasks.Task"));
             Assert.Equal(taskType, comp.Assembly.CorLibrary.GetTypeByMetadataName("System.Threading.Tasks.Task"));
         }
+
+        [WorkItem(863435, "DevDiv/Personal")]
+        [Fact]
+        public void CS1671ERR_BadModifiersOnNamespace01()
+        {
+            var test = @"
+public namespace NS // CS1671
+{
+    class Test
+    {
+        public static int Main()
+        {
+            return 1;
+        }
+    }
+}
+";
+            CreateCompilationWithMscorlib45(test).VerifyDiagnostics(
+                Diagnostic(ErrorCode.ERR_BadModifiersOnNamespace, "public"));
```

Comment:
>Diagnostic [](start = 16, length = 10)

Could you include the full diagnostic, including comment?
Same for test below.

Reply:
Done.